### PR TITLE
[3.2] cleos: add get supported_protocol_features and system activate subcommands

### PR DESCRIPTION
**Pull request by linh2931**
Backport https://github.com/EOSIO/eos/pull/10036 spiritually. Resolve https://github.com/eosnetworkfoundation/mandel/issues/462

Add `get supported_protocol_features` and `system activate` to `cleos` such that protocol features can be activated by using their names; no feature digests are needed.

The feature was totally rewritten. Hardcodied protocol feature names and digests are removed. User can use `get supported_protocol_features` to find a list of supported feature names. Command description was cleaned up and `PREACTIVATE_FEATURE` was removed as it is not possible to activate it using `activate` action from `eosio.boot` (RPC `schedule_protocol_feature_activations` must be used).

1. Find supported protocol features

```
./cleos get supported_protocol_features
ONLY_LINK_TO_EXISTING_PERMISSION
FORWARD_SETCODE
WTMSIG_BLOCK_SIGNATURES
GET_BLOCK_NUM
REPLACE_DEFERRED
NO_DUPLICATE_DEFERRED_ID
RAM_RESTRICTIONS
WEBAUTHN_KEY
BLOCKCHAIN_PARAMETERS
DISALLOW_EMPTY_PRODUCER_SCHEDULE
CRYPTO_PRIMITIVES
ONLY_BILL_FIRST_AUTHORIZER
RESTRICT_ACTION_TO_SELF
GET_CODE_HASH
ACTION_RETURN_VALUE
CONFIGURABLE_WASM_LIMITS2
FIX_LINKAUTH_RESTRICTION
GET_SENDER
```
2. Activate a protocol feature
```
./cleos system activate WTMSIG_BLOCK_SIGNATURES
executed transaction: 667479cd2280739b21b925c1c9cddc2d8dc46626f0ecc61173aec5f092431c61  128 bytes  179 us
#         eosio <= eosio::activate              {"feature_digest":"299dcb6af692324b899b39f16d5a530a33062804e41f09dc97e9f156b4476707"}
warn  2022-07-16T01:51:13.001 cleos     main.cpp:665                  print_result       warning: transaction executed locally, but may not be confirmed by the network yet
```

3. Activate an unknown protocol feature

```
./cleos system activate ABCD
ABCD is unknown. Following protocol features are supported

ONLY_LINK_TO_EXISTING_PERMISSION
FORWARD_SETCODE
WTMSIG_BLOCK_SIGNATURES
GET_BLOCK_NUM
REPLACE_DEFERRED
NO_DUPLICATE_DEFERRED_ID
RAM_RESTRICTIONS
WEBAUTHN_KEY
BLOCKCHAIN_PARAMETERS
DISALLOW_EMPTY_PRODUCER_SCHEDULE
CRYPTO_PRIMITIVES
ONLY_BILL_FIRST_AUTHORIZER
RESTRICT_ACTION_TO_SELF
GET_CODE_HASH
ACTION_RETURN_VALUE
CONFIGURABLE_WASM_LIMITS2
FIX_LINKAUTH_RESTRICTION
GET_SENDER
```

### CODE REVIEW

File: programs/cleos/main.cpp [hunk 1]
```diff
@@ -2247,6 +2249,76 @@ struct closerex_subcommand {
    }
 };
 
+struct activate_subcommand {
+   string feature_name_str;
+
+   activate_subcommand(CLI::App* actionRoot) {
+      auto activate = actionRoot->add_subcommand("activate", localized("Activate protocol feature by name"));
+      activate->add_option("feature",  feature_name_str,  localized("The name, one of below (lowercase also works):\n"
+                       "ONLY_LINK_TO_EXISTING_PERMISSION\n"
```
Comment (larryk85) [inline programs/cleos/main.cpp:2258]: Can we have a way of generating these from parsing the protocol_feature_manager.cpp? We have this pattern in quite a few places as is where we have hard-coded the protocol feature names and digests.  It ultimately makes it incredibly difficult to add new protocol features as all the places you have to go and fix are not connected and usually won't break certain tests, etc.

**Review by larryk85 — CHANGES_REQUESTED** [review]

File: programs/cleos/main.cpp [hunk 2]
```diff
@@ -2247,6 +2249,76 @@ struct closerex_subcommand {
...
+            {"GET_SENDER",                       "f0af56d2c5a48d60a4a5b5c903edfb7db3a736a94ed589d0b797df33ff9d3e1d"},
+            {"CRYPTO_PRIMITIVES",                "6bcb40a24e49c26d0a60513b6aeb8551d264e4717f306b81a37a5afb3b47cedc"},
+            {"GET_BLOCK_NUM",                    "35c2186cc36f7bb4aeaf4487b36e57039ccf45a9136aa856a5d569ecca55ef2b"}
+         };
+         // push system feature
+         string contract_account = "eosio";
+         string action="activate";
+         string data;
+         std::locale loc;
+         vector<std::string> permissions = {"eosio"};
```
Comment (heifner) [inline programs/cleos/main.cpp:2304]: Would be nice to provide an option to override the default permission.

File: programs/cleos/main.cpp (hunk 1 above)
Comment (linh2931) [reply programs/cleos/main.cpp:2258]: Thanks @larryk85 and @spoonincode I will call `get_supported_protocol_features` to get a list. This code will never change again when protocol features change.

File: programs/cleos/main.cpp (hunk 2 above)
Comment (linh2931) [reply programs/cleos/main.cpp:2304]: Thanks @heifner . Will add an option to provide more flexibility.

**Review by larryk85 — APPROVED** [review]

**Pull request merged**